### PR TITLE
adds testing

### PR DESCRIPTION
**Pull request by Jonny-Code**

### CODE REVIEW

File: angular/src/app/modules/account/account/account.component.spec.ts [hunk 1]
```diff
@@ -7,11 +7,12 @@ import { Account } from '../../../data/account.model';
 import { AccountService } from '../../../services/account/account.service';
 import { ACCOUNT_EDITING_SERVICE } from '../account-editing.token';
 import { ToastrService, ToastrModule } from 'ngx-toastr';
+import { toastrError } from '../../../utils/toastr/toastrError';
 
 describe('AccountComponent', () => {
   let toastrServiceSpy: jasmine.Spy;
-  const toastrSetvices = jasmine.createSpyObj('toastrSetvices', ['error']);
-  toastrServiceSpy = toastrSetvices.error.and.returnValue(of(''));
+  const toastrServiceStub = jasmine.createSpyObj('toastrServiceStub', ['error']);
```
Comment (escalonn) [inline angular/src/app/modules/account/account/account.component.spec.ts:14]: that's not what this first argument is for

(this file's comments also apply to the other 2 similar changesets)

File: angular/src/app/modules/account/account/account.component.spec.ts [hunk 2]
```diff
@@ -7,11 +7,12 @@ import { Account } from '../../../data/account.model';
...
+  toastrServiceSpy = toastrServiceStub.error.and.returnValue(of(''));
```
Comment (escalonn) [inline angular/src/app/modules/account/account/account.component.spec.ts:15]: that's not an error, [`throw`](https://www.learnrxjs.io/learn-rxjs/operators/creation/throw) is probably the right replacement for `of`

File: angular/src/app/modules/account/account/account.component.ts [hunk 3]
```diff
@@ -49,22 +49,19 @@ export class AccountComponent {
     this.profiles$ = this.account$.pipe(map((account) => account.profiles));
 
     // Pass initial model to editingService which acts as model for overwriting data coming in
+    this.callAccountGet(editingService);
+    // Register function for Payload release from editing service
+    editingService.payloadEmitter.subscribe((val) => this.update(val as Account));
+  }
+
+  callAccountGet(editingService: GenericEditingService<Partial<Account>>): void {
```
Comment (escalonn) [inline angular/src/app/modules/account/account/account.component.ts:57]: revert this

File: angular/src/app/modules/account/account/account.component.ts [hunk 4]
```diff
@@ -12,6 +12,7 @@ import { BookingService } from 'services/booking/booking.service';
 import { GenericEditingService } from 'services/editable/generic-editing.service';
 import { ACCOUNT_EDITING_SERVICE } from '../account-editing.token';
 import { ToastrService } from 'ngx-toastr'; // adding ngx-toastr for api service error notifications
+import { toastrError } from '../../../utils/toastr/toastrError';
```
Comment (escalonn) [inline angular/src/app/modules/account/account/account.component.ts:15]: make a toast service and encapsulate the ngx-toastr details and this util function in it.

File: angular/src/app/modules/lodging/lodging-details/lodging-details.component.spec.ts [hunk 5]
```diff
@@ -89,4 +139,47 @@ describe('LodgingDetailsComponent', () => {
...
+    score.setValue('1');
+    fixture.detectChanges();
+    expect(score.valid).toBeTrue();
+
+    message.setValue('');
+    fixture.detectChanges();
+    expect(message.valid).toBeFalse();
+    message.setValue('my message');
+    fixture.detectChanges();
+    expect(message.valid).toBeTrue();
```
Comment (escalonn) [inline angular/src/app/modules/lodging/lodging-details/lodging-details.component.spec.ts:172]: ideally one behavior gets one spec, but it's borderline

**Review by escalonn — CHANGES_REQUESTED** [review]

**Review by markmoorerev — APPROVED** [review]

File: angular/src/app/modules/account/account/account.component.spec.ts [hunk 6]
```diff
@@ -72,9 +62,26 @@ describe('AccountComponent', () => {
...
+   * tests if bad request calls toastservice
+   */
+  it('should throw error and call toast error', () => {
+    accountServiceStub.getWithError().subscribe({
+      error: (err) => {
+        expect(err).toEqual({ error: 'Test error' });
+      },
+    });
+    expect(toastServiceSpy.toastError).toHaveBeenCalled();
+    toastServiceSpy.toastError.calls.reset();
```
Comment (escalonn) [inline angular/src/app/modules/account/account/account.component.spec.ts:85]: no... this is supposed to be a test of the component, but the component plays no role in what you've written here. you're just testing stubs against stubs.

the _test_ shouldn't be calling the `get`. the component is already calling the `get`. your job is to provide an object to the component which will throwError when someone calls `get` on it, in this test. in the other test, it should provide an actual good result.

**Review by escalonn — CHANGES_REQUESTED** [review]

**Comment by fredbelotte** [comment]
inactive

**Pull request closed without merging**